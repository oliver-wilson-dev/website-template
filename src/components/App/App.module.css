@import "../../styles/colours.css";

.App {
  text-align: center;
  display: flex;
  flex-direction: column;
  min-height: 100vh;
  justify-content: flex-end;
  background-color: var(--background-colour);

  & div, p, h1, h2, h3, h4, h5, span, a {
    transition: background-color 500ms ease-in-out, color 1000ms ease-in-out;
  }
}

a {
  text-decoration: none;
  color: $link-blue;
}

p {
  font-size: 14px;
}

p, h1, h2, h3, h4, h5, span, a {
  color: var(--text-colour);
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
