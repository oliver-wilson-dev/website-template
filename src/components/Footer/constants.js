export const GITHUB_PROFILE_URL = 'https://github.com/oliver-wilson-dev/';
export const PHONE_NUMBER = '07580081310';
export const EMAIL_ADDRESS = 'contact@oliverwilson.dev';
