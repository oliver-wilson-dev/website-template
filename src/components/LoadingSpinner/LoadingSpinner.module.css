@import "../../styles/colours.css";

.wrapper {
  display: flex;
  justify-content: center;
  align-items: center;
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  background: $dark-grey;
  opacity: 0.5;

  visibility: hidden;
  opacity: 0;
  pointer-events: none;
  transition: all 0.25s ease-out;

  &Show {
    visibility: visible;
    opacity: 0.2;
    pointer-events: auto;
  }

  &Hide {
    transition-delay: 0.25s;
    visibility: hidden;
    opacity: 0;
    pointer-events: none;
  }
}

.loadingSpinner {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-grow: 1;
  transform: scale(0);
  transition: all 0.25s ease-out;
  position: fixed;
  top: 50%;
  left: 50%;

  &Show {
    transform: scale(1);
  }

  &Hide {
    transform: scale(0);
  }
}

.loadingSpinner div {
  box-sizing: border-box;
  display: block;
  position: absolute;
  width: 100px;
  height: 100px;
  margin: 8px;
  border: 8px solid var(--background-colour-inverse);
  border-radius: 50%;
  animation: spin 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
  border-color: var(--background-colour-inverse) transparent transparent transparent;
}

.loadingSpinner div:nth-child(1) {
  animation-delay: -0.45s;
}

.loadingSpinner div:nth-child(2) {
  animation-delay: -0.3s;
}

.loadingSpinner div:nth-child(3) {
  animation-delay: -0.15s;
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
