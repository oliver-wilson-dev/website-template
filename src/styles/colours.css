
$background-grey: #e9ecef;

$dark-grey:  #333333;

$text-dark-grey:  #333333;
$text-light-grey:  #ffffff;

$link-blue: #1540cc;
$link-yellow: #fcbf49;

$nav-background: #f8f8f8;