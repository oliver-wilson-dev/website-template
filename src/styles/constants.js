export const LIGHT_THEME = Symbol('LIGHT_THEME');
export const DARK_THEME = Symbol('DARK_THEME');
