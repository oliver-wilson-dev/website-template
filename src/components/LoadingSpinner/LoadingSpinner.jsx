/*
  eslint-disable
  import/no-unresolved,
  jsx-a11y/click-events-have-key-events,
  jsx-a11y/no-static-element-interactions
*/
import React from 'react';
import cn from 'classnames';

import styles from './LoadingSpinner.module.css';

const onClick = (e) => {
  e.stopPropagation();
};

const LoadingSpinner = ({ show }) => (
  <>
    <div
      className={cn(styles.wrapper,
        {
          [styles.wrapperShow]: show,
          [styles.wrapperHide]: !show
        })}
      onClick={onClick}
    />
    <div className={cn(styles.loadingSpinner,
      {
        [styles.loadingSpinnerShow]: show,
        [styles.loadingSpinnerHide]: !show
      })}
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </>
);

export default LoadingSpinner;
