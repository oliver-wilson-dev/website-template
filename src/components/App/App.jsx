import React, { Suspense, useState } from 'react';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import cn from 'classnames';

import Navigation from '../Navigation';
import Footer from '../Footer';
import LoadingSpinner from '../LoadingSpinner';
import Switch from '../Switch';

import styles from './App.module.css';
import themeStyles from '../../styles/theme.module.css';
import Fallback from '../Fallback';
import CookieDisclaimer from '../../containers/CookieDisclaimer';
import ErrorBoundary from '../ErrorBoundary';
import { LIGHT_THEME, DARK_THEME } from '../../styles/constants';

const App = ({ theme }) => {
  const [loading, setLoading] = useState(false);

  return (
    <div className={cn({
      [themeStyles.light]: theme === LIGHT_THEME,
      [themeStyles.dark]: theme === DARK_THEME
    }, styles.App)}
    >
      <Router>
        <Navigation />
        <LoadingSpinner show={loading} />
        <Suspense fallback={<Fallback setLoading={setLoading} />}>
          <ErrorBoundary>
            <Switch />
          </ErrorBoundary>
        </Suspense>
        <CookieDisclaimer />
        <Footer />
      </Router>
    </div>
  );
};

export default App;
