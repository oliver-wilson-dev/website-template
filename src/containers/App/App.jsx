import { connect } from 'react-redux';
import App from '../../components/App';
import { getTheme } from '../../state/selectors';

const mapStateToProps = (state) => ({
  theme: getTheme({ state }),
});

export default connect(mapStateToProps)(App);
