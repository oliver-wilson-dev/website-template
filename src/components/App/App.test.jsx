import React from 'react';
import { shallow } from 'enzyme';
import LoadingSpinner from '../LoadingSpinner';
import App from './App';
import { LIGHT_THEME, DARK_THEME } from '../../styles/constants';

jest.mock('react-router-dom', () => ({
  __esModule: true,
  BrowserRouter: ({ children }) => <>{children}</>,
  Route: ({ children }) => <>{children}</>,
  Switch: ({ children }) => <>{children}</>
}));

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  Suspense: ({ children, fallback }) => (
    <>
      {fallback()}
      {children}
    </>
  ),
}));

jest.mock('../ErrorBoundary', () => {
  const ErrorBoundary = ({ children }) => <>{children}</>;

  return ErrorBoundary;
});

jest.mock('../Navigation', () => {
  const Navigation = () => null;

  return Navigation;
});

jest.mock('../Fallback', () => {
  const Fallback = () => null;

  return Fallback;
});

jest.mock('../LoadingSpinner', () => {
  const LoadingSpinner = () => null;

  return LoadingSpinner;
});

jest.mock('../Footer', () => {
  const Footer = () => <footer />;

  return Footer;
});

jest.mock('../../containers/CookieDisclaimer', () => {
  const CookieDisclaimer = () => null;

  return CookieDisclaimer;
});

const HomePage = () => null;
const AboutPage = () => null;
const ContactPage = () => null;

jest.mock('../../routes', () => ({
  home: {
    component: HomePage,
    route: 'test-home-route'
  },
  about: {
    component: AboutPage,
    route: 'test-about-route'
  },
  contact: {
    component: ContactPage,
    route: 'test-contact-route'
  }
}));

const render = (props = {}) => shallow(<App {...props} />);

describe('App', () => {
  describe('when the theme is the light theme', () => {
    it('renders correctly', () => {
      expect(render({ theme: LIGHT_THEME })).toMatchSnapshot();
    });
  });

  describe('when the theme is the dark theme', () => {
    it('renders correctly', () => {
      expect(render({ theme: DARK_THEME })).toMatchSnapshot();
    });
  });

  it('should call the LoadingSpinner with the show prop set to false', () => {
    const wrapper = render();
    expect(wrapper.find(LoadingSpinner).props()).toEqual({ show: false });
  });
});
