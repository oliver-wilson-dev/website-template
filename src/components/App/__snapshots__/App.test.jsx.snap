// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App when the theme is the dark theme renders correctly 1`] = `
<div
  className="dark App"
>
  <BrowserRouter>
    <Navigation />
    <LoadingSpinner
      show={false}
    />
    <Suspense
      fallback={
        <Fallback
          setLoading={[Function]}
        />
      }
    >
      <ErrorBoundary>
        <Switch />
      </ErrorBoundary>
    </Suspense>
    <CookieDisclaimer />
    <Footer />
  </BrowserRouter>
</div>
`;

exports[`App when the theme is the light theme renders correctly 1`] = `
<div
  className="light App"
>
  <BrowserRouter>
    <Navigation />
    <LoadingSpinner
      show={false}
    />
    <Suspense
      fallback={
        <Fallback
          setLoading={[Function]}
        />
      }
    >
      <ErrorBoundary>
        <Switch />
      </ErrorBoundary>
    </Suspense>
    <CookieDisclaimer />
    <Footer />
  </BrowserRouter>
</div>
`;
